### DISCUSSION TITLE
PDF parser in Cordova + React

### DISCUSSION THREAD

**Discussion by marek8623** [apache/cordova#390] [Q&A]
Hi everyone,

I am looking for solution to parse PDF and get text and images (not viewing but using data from PDF).
The project is based on cordova and react (cordova ios and cordova electron) + webpack.

For getting the file I am already using [cordova-plugin-file](https://cordova.apache.org/docs/en/11.x/reference/cordova-plugin-file/). I tried some of the plugins for parsing PDF data like [pdfreader](https://www.npmjs.com/package/pdfreader) or [pdf-parse](https://www.npmjs.com/package/pdf-parse).

I also tried [npm-pdfreader](https://github.com/adrienjoly/npm-pdfreader) but I am getting following error:

```
ERROR in ./node_modules/pdfreader/index.js 3:9
Module parse failed: Unexpected token (3:9)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export { PdfReader } from "./PdfReader.js";
| export { Rule } from "./Rule.js";
> export * as LOG from "./lib/LOG.js";
| import * as parseTableExports from "./lib/parseTable.js";
```

So my question is if someone can help me how to parse PDF together with [cordova-plugin-file](https://cordova.apache.org/docs/en/11.x/reference/cordova-plugin-file/). Or is there some cordova plugin that I can use?

Thanks a lot for help and guidance.

**Comment by breautek** [comment]
> ERROR in ./node_modules/pdfreader/index.js 3:9
Module parse failed: Unexpected token (3:9)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders

This signals that Webpack failed to read `node_modules/pdfreader/index.js`

It may be using a newer JS syntax than what webpack supports. There's not enough information here to say with in certainty, but I'd recommend ensuring you're using the latest webpack.

Note that those PDF modules might also not be developed to be ran in the browser. I haven't looked at them specifically, but if they rely purely on NodeJS APIs like buffers and streams, you will have a difficult time to get the module working in a browser/cordova environment, if at all.

Ideally they should be able to accept a browser's `Blob`/`ArrayBuffer` APIs.
